js/runtime.js

```diff
@@ -78,38 +82,46 @@
 
     var perFrameHandlers;
     var lastTime;
+    var animationFrameHandler;
 
     function clearPerFrameHandlers() {
         perFrameHandlers = [];
-        lastTime = new Date().valueOf();
+        if (animationFrameHandler){
+            cancelAnimationFrame(animationFrameHandler);
+            animationFrameHandler = null;
+        }
     }
 
     // Initialize the stage.
     Event.on(window, 'ui:resize', null, handleResize);
     Event.on(document.body, 'ui:wb-resize', null, handleResize);
 
-
     function startEventLoop(){
         clearPerFrameHandlers();
+        lastTime = new Date().valueOf();
+        runtime.control._startTime = lastTime;
         runtime.control._frame = 0;
         runtime.control._sinceLastTick = 0;
-        requestAnimationFrame(frameHandler);
+        animationFrameHandler = requestAnimationFrame(frameHandler);
     }
 
     function stopEventLoop() {
         /* TODO: Dunno lol there be more in here? */
+        // console.log('stop event loop');
     }
 
-    function frameHandler(){
+    function frameHandler(timestamp){
         // where to put these? Event already has some global state.
-        var currTime = new Date().valueOf();
-        runtime.control._elapsed = currTime - lastTime;
+        runtime.control._elapsed = timestamp - runtime.control._startTime;
```

Comment:
Animation frame handlers get a timestamp already as their only argument, we don't need to create one ourselves (and the one they get is higher precision than what we get from new Date() anyway).